### DISCUSSION TITLE
check particles on boundary in HemoCellParticleField::advanceParticles()

### DISCUSSION THREAD

**Discussion by mojtabaAmiraslanpour** [UvaCsl/HemoCell#109] [General discussion]
Hi,

I am looking at the advanceParticles function and first of all I don't get why it is subtracting location from particle position. Second, I printed out x, y, z and the bounding box and couldn't understand why there are negative values for x, y?

```
         x, y, z: {-16,29,10}
         Location: {0,0,0}
         Location: {43,43,23}

         x, y, z: {-17,27,9}
         Location: {0,0,0}
         Location: {43,43,23} 

         x, y, z: {-12,24,13}
         Location: {0,0,0}
         Location: {43,43,23}
...
```

Can you elaborate on this a little please?

**Comment by gzavo** [comment]
I'm not sure what you are asking here.
Do you mean this function? https://github.com/UvaCsl/HemoCell/blob/d91698721f7a06166cb6ac8d22fa2f2f4baa4ed7/core/hemoCellParticle.h#L188

Maybe the negative values are in the communication envelope if you have periodicity in x?
